phaser3 rewrite

sdairs opened the pull request:
The version of Phaser is quite old, so I started migrating it to Phaser3. It's pretty much all working, but still need some polish.

Also brings in Phaser via npm & uses vite

sdairs commented:
Added:

- Tinybird data project files with a top 10 sessions API
- Basic JS fetch to send individual point scores to Tinybird Events API

JoeKarlsson commented on src/MainMenuScene.js:
```diff
@@ -0,0 +1,28 @@
+    });
+
+    graphics.on(
+      "pointerup",
```
We should add both keyboard press up, and touch input anywhere on the screen to start the next round. Feels clunky to have to switch to the mouse to start the next round.

sdairs replied:
👍 added

JoeKarlsson commented on src/FlappyTinybirdScene.js:
```diff
@@ -0,0 +1,105 @@
+      Phaser.Input.Keyboard.KeyCodes.SPACE
+    );
+
+    spaceKey.on("down", (key, event) => {
```
We should make this mobile friendly. I added `this.input.onDown.add(this.jump, this);` in my old version to get this to work on mobile.

sdairs replied:
👍 added

JoeKarlsson reviewed (approved):
This rewrite looks amazing. Nice work. Love how you were able to make the game easier to play ;) I have some minor changes that would be good to fix, but we can totally handle those in future PRs since this one is starting to get pretty big. 

Also, looks like Phaser debugger mode is still on too ;)

sdairs commented:
1. Added touch input as well as space to jump.
2. Added a HTML form to input a name & email address for the scoreboard. You cannot press play without entering details.
3. Changed it so that the app container is sized & centered, rather than the canvas, because the DOM overlay is placed inside the same container at the same level as the canvas, so it was off-center

TODO:

Need to style the HTML form!
An 'end game' screen would be good, that shows your score, and gives you the option to 'Play again' without having to re-enter your name/email

sdairs commented:
Top 10 Scoreboard API + read only auth token

https://api.tinybird.co/v0/pipes/top_10_sessions.json?token=p.eyJ1IjogIjYwOTQyZDQ5LTU1NzYtNDIyMC04YTRjLTZhOTVlMmU1N2RlNCIsICJpZCI6ICI1ODRmMWJhZi1lZTc4LTQ5MmUtOWIxYS1iZjcyNjk4MGFkZjAifQ.xZJUmNrN0f9MFO91-i-eA9_iJH25nFYeNruTn55SqQI
